### ISSUE TITLE
Treat abstention votes differently from nay votes

### ISSUE THREAD

**Issue by taoeffect** [okTurtles/group-income#596]
### Problem

An 80% threshold might not be the clearest way to represent voting to users.

### Solution

We might want to think in terms of nay votes (objections) instead of in terms of thresholds. Design this voting system and user test it against threshold-based voting. In both kinds of voting abstention votes should be treated differently from nay votes, and this needs to be clearly expressed in the UI/UX.

Note that this issue is both about the code (implementation) and the UI.

References:

- https://groupincome.org/2016/06/what-makes-a-good-voting-system/
- https://groupincome.org/2016/09/misconceptions-about-majority-rule/

**Comment by taoeffect** [comment]
Removing myself from the assignees because the implementation (code) part of this has been done in #611.

**Comment by taoeffect** [comment]
Copying text of #878 (duplicate):

### Problem

Currently you can only vote "yes" or "no" on a proposal, but cannot indicate that you don't care. A non-vote is not the same as a vote of not caring. Our `RULE_THRESHOLD` supports the math for handling `VOTE_INDIFFERENT`, but our UI does not.

### Solution

Implement in UI a way to indicate indifference about the outcome of a proposal.

**Comment by mmbotelho** [comment]
Note: This comment is outdated. [Go to the updated comment](https://github.com/okTurtles/group-income-simple/issues/596#issuecomment-626357834)

-------

Added an option to vote blank. You will notice that there's a new link style, in gray. I plan on adding this link to our design system and turn some of the links we already have into this more subdued version. This will help improve our visual hierarchy, the app is becoming spotted with blue links and that can be very "loud" visually:

<img width="596" alt="Captura de ecrã 2020-04-30, às 12 42 32" src="https://user-images.githubusercontent.com/22004566/80708325-654d5e80-8ae3-11ea-993c-eda71b120207.png">

The mobile version will now open a select with the options to vote yes, vote no or vote blank since 3 actions are too much to display at once in a smaller screen:

<img width="381" alt="Captura de ecrã 2020-04-30, às 13 05 17" src="https://user-images.githubusercontent.com/22004566/80709341-0be62f00-8ae5-11ea-94dc-772262f94a2c.png">

**Comment by sandrina-p** [comment]
@mmbotelho do you mean opening an HTML `<select>` with 3 options when we click "Vote"? If yes, then you're suggesting to automatically "dispatch" the vote action once the user closes the select element?
Hmm... It's too bad we can't actually test that behavior in an actual phone because of #751. I'm afraid that behavior would not work properly. When I implement this I'll try to replicate it on codepen to make sure it works as you expect.

**Comment by mmbotelho** [comment]
> then you're suggesting to automatically "dispatch" the vote action once the user closes the select element?

I'm not sure if I understand what you're saying here. A vote is only recorded if the user explicitly selected one of the options. If they close the select without choosing an option, no vote is recorded. Is that what you were saying?

**Comment by sandrina-p** [comment]
I'll need to verify how the event "change" is dispatched inside a select on mobile (close vs select vs select + done). I'll get back to you later this week about this.

**Comment by mmbotelho** [comment]
Ok, thank you @sandrina-p! We can find a different solution if this one doesn't work

**Comment by sandrina-p** [comment]
Hi @mmbotelho. Here's the Codepen example: https://codepen.io/sandrina-p/full/KKdXmxd. I hope you better understand what's the problem with your approach.

**Comment by mmbotelho** [comment]
Thank you for sharing that, I will design a custom drawer with a save button then. I'll post it here when I have it!

**Comment by taoeffect** [comment]
@mmbotelho I like the design!

Only critique is "vote blank" doesn't mean much to much, so as a user I would find that language confusing. I would prefer the clarity of something like "Vote indifferent".

**Comment by mmbotelho** [comment]
Ok @taoeffect , I can change the copy! Blank vote is a very popular concept here, that's why I used it, but maybe it doesn't translate very well!

@sandrina-p Idesigned this component for this:
<img width="263" alt="Captura de ecrã 2020-05-06, às 16 50 56" src="https://user-images.githubusercontent.com/22004566/81199160-368b2880-8fba-11ea-92cf-19e2dec2bcf3.png">

**Comment by sandrina-p** [comment]
Is that a new component, right? Could you add the figma link with the detailed specs?

**Comment by mmbotelho** [comment]
Yes, sorry I forgot to link it! [Mobile dropdown on Figma](https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=2870%3A7)

**Comment by sandrina-p** [comment]
The component looks okay. Could you also share the Figma example showing the voting options in this component? I didn't find it on Figma - Design System page :/

**Comment by mmbotelho** [comment]
[My previous comment has the behavior explained](https://github.com/okTurtles/group-income-simple/issues/596#issuecomment-621798488). Is that enough or do you need an actual mockup of it?

**Comment by mmbotelho** [comment]
I'll make one @sandrina-p

**Comment by mmbotelho** [comment]
Updated version with the feedback on this thread:

Added an option to **vote indifferent**. You will notice that there's a new link style, in gray ([Figma specs here](https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=3070%3A106)). I plan on turning some of the links we already have into this more subdued version. This will help improve our visual hierarchy, the app is becoming spotted with blue links and that can be very "loud" visually:

<img width="929" alt="Captura de ecrã 2020-05-10, às 17 49 12" src="https://user-images.githubusercontent.com/22004566/81505375-a9601080-92e6-11ea-9450-90608443ccbb.png">

The mobile version will now open a bottom action sheet with the options to vote yes, vote no or vote blank since 3 actions are too much to display at once in a smaller screen:

<img width="1006" alt="Captura de ecrã 2020-05-10, às 17 58 04" src="https://user-images.githubusercontent.com/22004566/81505534-cfd27b80-92e7-11ea-8d47-aec160c57e2b.png">

Figma links:
- [Bottom action sheet on the design system](https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=2870%3A7);
- [Flow to vote, mobile](https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=2119%3A101);

**Comment by sandrina-p** [comment]
The flow to vote (mobile) is great because it shows the order of voting (which is different from desktop) Thanks, @mmbotelho!
